lib/model/model.go

```diff
@@ -1823,21 +1809,22 @@ func (m *model) AddConnection(conn connections.Connection, hello protocol.HelloR
 		return
 	}
 
-	m.pmut.Lock()
-	if oldConn, ok := m.conn[deviceID]; ok {
+	m.pmut.RLock()
```

Comment:
I think both the new and old code is incorrect here. If there is an old connection we will attempt to close it. But since we don't hold the lock over the close, and don't remove the connection from `m.conn` we can have two concurrent calls both seeing the old conn and closing it, then setting the new conn with the other call clobbering it...

Reply:
In principle yes, in practice this should be prevented by `AddConnection` being called from a single loop in `lib/service`, thus this cannot be called concurrently.